src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,42 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    static final int BOUND = 100;
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+    private final Random random = new Random();
+
+    public Figure getRandomFigure() {
+        int index = random.nextInt(FiguresList.values().length);
+        FiguresList randomFigure = FiguresList.values()[index];
+        switch (randomFigure) {
+            case CIRCLE:
+                return new Circle(String.valueOf(colorSupplier.getRandomColor()),
```

Comment:
1. String representation of enum is obtainable by calling .name() method
2. I'd recommend to make colorSupplier.getRandomColor() to return String instead of Color not to convert in every usage
```suggestion
                return new Circle(colorSupplier.getRandomColor(),
```